torch/_inductor/ir.py

```diff
@@ -5638,9 +5638,10 @@ def create(cls, kernel, *args, **kwargs):
                 unbacked_bindings,
             ) = cls.process_kernel(kernel, *args, **kwargs)
 
+        device = cls.find_device(tensor_args, example_output)
         if example_output is None:
             packed = cls(
-                NoneLayout(None),
+                NoneLayout(device),
```

Comment:
Now that we're propagating device here, can we remove some of the `if device is not None` checks from the original pr:


https://github.com/pytorch/pytorch/commit/493478db4ac69179c6c8a870d573656469fcad1e#diff-d2539c9c8dc6a3d7e457767a880612e96d3c85752a77ead49a9e4e00a3e4c3c7R2443

Reply:
From the implementation of `FallbackKernel.find_device`, https://github.com/pytorch/pytorch/blob/685fcfb40de1bba6a99b8dc454a99ffc4b44f9a9/torch/_inductor/ir.py#L5455-L5471, I think the device may still be `None`.